classes/citation/CitationListTokenizerFilter.inc.php

```diff
@@ -36,18 +36,65 @@ function CitationListTokenizerFilter() {
 	 */
 	function &process(&$input) {
 		// The default implementation assumes that raw citations are
-		// separated with line endings.
-		// 1) Remove empty lines and normalize line endings.
-		$input = String::regexp_replace('/[\r\n]+/s', "\n", $input);
-		// 2) Remove trailing/leading line breaks.
+		// separated with a blank line.
+
+		// Normalize line endings.
+		if (strpos($input, "\r") !== FALSE) {
```

Comment:
Maybe with high ASCII and/or non-UTF-8 multibyte encodings, but [legit UTF-8 should be fine](https://stackoverflow.com/questions/2652193/can-str-replace-be-safely-used-on-a-utf-8-encoded-string-if-its-only-given-vali).

I can see the value in using the standard `PKPString` functions everywhere, though.